iris-fsw-softconsole/src/drivers/protocol/spi.c

```
@@ -117,20 +122,29 @@ void spi_transaction_block_read_with_toggle(CoreSPIInstance_t core, spi_slave_t
 //-------------------------------------------------------------------------------------------------------------------------------------------------------------
 void spi_transaction_block_write_without_toggle(CoreSPIInstance_t core, spi_slave_t slave, mss_gpio_id_t pin, uint8_t * cmd_buffer, uint16_t cmd_size, uint8_t * wr_buffer, uint16_t wr_size)
 {
-	//TODO: Should we use a static buffer instead of malloc? If not we need to handle if malloc fails.
 
 	//Put the command and data into one buffer.
 	uint32_t total_count = cmd_size + wr_size;
-	uint8_t * buffer = pvPortMalloc(total_count);
-
-	memcpy(buffer,cmd_buffer,cmd_size);
-	memcpy(&buffer[cmd_size],wr_buffer,wr_size);
-
-	//Select the slave and then perform SPI transfer.
-    SPI_set_slave_select(&core_spi[core], slave);
-    SPI_transfer_block(&core_spi[core],buffer, total_count, 0, 0);
-    SPI_clear_slave_select(&core_spi[core],slave);
 
+	if(total_count < SPI_BUFF_SIZE){
+
+		//Copy the data into the static buffer.
+		//We should not need to clear the buffer since we overwrite previous data and know the length.
+        memcpy(spi_temp_buff,cmd_buffer,cmd_size);
+        memcpy(&spi_temp_buff[cmd_size],wr_buffer,wr_size);
+
+        //Select the slave and then perform SPI transfer.
+        SPI_set_slave_select(&core_spi[core], slave);
+        SPI_transfer_block(&core_spi[core],spi_temp_buff, total_count, 0, 0);
+        SPI_clear_slave_select(&core_spi[core],slave);
+
+	}
+	else{
+		//Handle when the data exceeds the size of the static buffer.
+		//Shouldn't really happen since we can increase size of the buffer
+		//during development or modify the calling code..
+	    while(1){}// Just loop here for now, until we add error code.
```

Is there a better option instead of looping?

---

Yeah, we should return an error code in the actual production code, since we wouldn't want the whole software to hang. 

We could add an error code now, but we will have to change all the code that uses SPI to handle the error and we haven't fully decided on how the error codes/ error handling will work yet... 

But for now looping will make it very obvious when we are trying to transmit more data than fits in the buffer, so we can then fix the problem.